src/PersonObjects/Player/PlayerObjectGeneralMethods.tsx

```diff
@@ -1764,66 +1756,44 @@ export function applyForJob(this: IPlayer, entryPosType: CompanyPosition, sing =
   let pos = entryPosType;
 
   if (!this.isQualified(company, pos)) {
-    const reqText = getJobRequirementText(company, pos);
     if (!sing) {
-      dialogBoxCreate("Unfortunately, you do not qualify for this position<br>" + reqText);
+      dialogBoxCreate("Unfortunately, you do not qualify for this position<br>" + getJobRequirementText(company, pos));
     }
     return false;
   }
 
-  // Check if this company has the position
   if (!company.hasPosition(pos)) {
+    console.error(`Company ${company.name} does not have position ${pos}. Player.applyToCompany() failed`);
     return false;
   }
 
   while (true) {
-    const newPos = getNextCompanyPositionHelper(pos);
-    if (newPos == null) {
-      break;
-    }
-
-    //Check if this company has this position
-    if (company.hasPosition(newPos)) {
-      if (!this.isQualified(company, newPos)) {
-        //If player not qualified for next job, break loop so player will be given current job
-        break;
-      }
-      pos = newPos;
-    } else {
-      break;
-    }
+    const nextPos = getNextCompanyPositionHelper(pos);
+    if (nextPos == null) break;
+    if (company.hasPosition(nextPos) && this.isQualified(company, nextPos)) {
+      pos = nextPos;
+    } else break;
   }
 
-  //Check if the determined job is the same as the player's current job
-  if (currCompany != null) {
-    if (currCompany.name == company.name && pos.name == currPositionName) {
+  //Check if player already has the assigned job
+  if (this.jobs[company.name] === pos.name) {
+    if (!sing) {
       const nextPos = getNextCompanyPositionHelper(pos);
-      if (nextPos == null) {
-        if (!sing) {
-          dialogBoxCreate("You are already at the highest position for your field! No promotion available");
-        }
-        return false;
-      } else if (company.hasPosition(nextPos)) {
-        if (!sing) {
-          const reqText = getJobRequirementText(company, nextPos);
-          dialogBoxCreate("Unfortunately, you do not qualify for a promotion<br>" + reqText);
-        }
-        return false;
+      if (nextPos == null || !company.hasPosition(nextPos)) {
+        dialogBoxCreate("You are already at the highest position for your field! No promotion available");
       } else {
-        if (!sing) {
-          dialogBoxCreate("You are already at the highest position for your field! No promotion available");
-        }
-        return false;
+        const reqText = getJobRequirementText(company, nextPos);
+        dialogBoxCreate("Unfortunately, you do not qualify for a promotion<br>" + reqText);
       }
     }
+    return false;
   }
 
   this.jobs[company.name] = pos.name;
-  if (!this.focus && this.isWorking && this.companyName !== this.location) this.resetWorkStatus();
```

Comment:
hmm but if we dont call this wont we run into issues with the workness of the user? or is that intentional as in carry over time worked into the new job? imo shouldn't we finish their work and start a new work session

Reply:
If player is currently working a shift at company X and applies for a new job at company Y, there's no need to reset work status. We can just add the newly applied for job without interrupting the current work shift.

We do have to reset the work status when the player decides to start actually _working_ at company Y, but that's handled by a separate function. This function is just for applying for a new job which has no reason to reset work status.

It used to need to do this when the player could only have one job - if the player applied for a new job we would need to cancel any in progress job at a different company because the player's employer was being replaced. But now with multiple jobs, nothing is being replaced when the player applies at a new company.